intranet/apps/eighth/views/admin/scheduling.py

```diff
@@ -101,6 +102,10 @@ def schedule_activity_view(request):
                     for field_name in fields:
                         obj = form.cleaned_data[field_name]
                         logger.debug("%s %s", field_name, obj)
+
+                        if field_name == "rooms" and (not instance.is_both_blocks or instance.block.block_letter == "A"):
```

Comment:
It seems that it passes this condition even when the field does not change.

Removing my approval.